src/shell.js

```diff
@@ -193,11 +193,25 @@ if (ENVIRONMENT_IS_NODE) {
   if (typeof process == 'undefined' || !process.release || process.release.name !== 'node') throw new Error('not compiled for this environment (did you build to HTML and try to run it not on the web, or set ENVIRONMENT to something - like node - and run it someplace else - like on the web?)');
 #endif
 #endif
+  // These modules will usually be used on Node.js. Load them eagerly to avoid
+  // the complexity of lazy-loading.
+  var fs = require('fs');
+  var nodePath = require('path');
```

Comment:
In moving these require statements we've also lost that long comment explaining them.  Was the deliberate?

Reply:
That was intentional. That comment mentioned that it was building an ES6 module where `require()` did not exist. This (draft) PR would solve that issue.

We could argue for excluding these imports when building with `-sSINGLE_FILE`. But in that case, `stdin` usage, or using the `-sMAIN_MODULE` and/or `-sNODERAWFS` options would still depend on these imports (as noted in PR https://github.com/emscripten-core/emscripten/pull/17851).